rag_app/ingest.py

```diff
@@ -13,7 +17,13 @@ def read_files(path: Path, file_suffix: str) -> Iterable[Document]:
     for i, file in enumerate(path.iterdir()):
         if file.suffix != file_suffix:
             continue
-        yield Document(id=i, text=file.read_text(), filename=file.name)
+        post = frontmatter.load(file)
```

Comment:
the `post.metadata` variable is a dictionary mapping of the entire frontmatter definition